File: Server/src/main/java/com/api/server/foodb/Compound.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "compounds")
public class Compound {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "legacy_id")
    private Integer legacyId;

    @Size(max = 255)
    @NotNull
    @Column(name = "type", nullable = false)
    private String type;

    @Size(max = 9)
    @NotNull
    @Column(name = "public_id", nullable = false, length = 9)
    private String publicId;

    @Size(max = 255)
    @NotNull
    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "export")
    private Boolean export;

    @Size(max = 255)
    @Column(name = "state")
    private String state;

    @Size(max = 255)
    @Column(name = "annotation_quality")
    private String annotationQuality;

    @Lob
    @Column(name = "description")
    private String description;

    @Size(max = 255)
    @Column(name = "cas_number")
    private String casNumber;

    @Lob
    @Column(name = "melting_point")
    private String meltingPoint;

    @Size(max = 255)
    @Column(name = "protein_formula")
    private String proteinFormula;

    @Size(max = 255)
    @Column(name = "protein_weight")
    private String proteinWeight;

    @Size(max = 255)
    @Column(name = "experimental_solubility")
    private String experimentalSolubility;

    @Size(max = 255)
    @Column(name = "experimental_logp")
    private String experimentalLogp;

    @Size(max = 255)
    @Column(name = "hydrophobicity")
    private String hydrophobicity;

    @Size(max = 255)
    @Column(name = "isoelectric_point")
    private String isoelectricPoint;

    @Lob
    @Column(name = "metabolism")
    private String metabolism;

    @Size(max = 255)
    @Column(name = "kegg_compound_id")
    private String keggCompoundId;

    @Size(max = 255)
    @Column(name = "pubchem_compound_id")
    private String pubchemCompoundId;

    @Size(max = 255)
    @Column(name = "pubchem_substance_id")
    private String pubchemSubstanceId;

    @Size(max = 255)
    @Column(name = "chebi_id")
    private String chebiId;

    @Size(max = 255)
    @Column(name = "het_id")
    private String hetId;

    @Size(max = 255)
    @Column(name = "uniprot_id")
    private String uniprotId;

    @Size(max = 255)
    @Column(name = "uniprot_name")
    private String uniprotName;

    @Size(max = 255)
    @Column(name = "genbank_id")
    private String genbankId;

    @Size(max = 255)
    @Column(name = "wikipedia_id")
    private String wikipediaId;

    @Lob
    @Column(name = "synthesis_citations")
    private String synthesisCitations;

    @Lob
    @Column(name = "general_citations")
    private String generalCitations;

    @Lob
    @Column(name = "comments")
    private String comments;

    @Size(max = 255)
    @Column(name = "protein_structure_file_name")
    private String proteinStructureFileName;

    @Size(max = 255)
    @Column(name = "protein_structure_content_type")
    private String proteinStructureContentType;

    @Column(name = "protein_structure_file_size")
    private Integer proteinStructureFileSize;

    @Column(name = "protein_structure_updated_at")
    private Instant proteinStructureUpdatedAt;

    @Size(max = 255)
    @Column(name = "msds_file_name")
    private String msdsFileName;

    @Size(max = 255)
    @Column(name = "msds_content_type")
    private String msdsContentType;

    @Column(name = "msds_file_size")
    private Integer msdsFileSize;

    @Column(name = "msds_updated_at")
    private Instant msdsUpdatedAt;

    @Column(name = "creator_id")
    private Integer creatorId;

    @Column(name = "updater_id")
    private Integer updaterId;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

    @Column(name = "phenolexplorer_id")
    private Integer phenolexplorerId;

    @Size(max = 255)
    @Column(name = "dfc_id")
    private String dfcId;

    @Size(max = 255)
    @Column(name = "hmdb_id")
    private String hmdbId;

    @Size(max = 255)
    @Column(name = "duke_id")
    private String dukeId;

    @Size(max = 255)
    @Column(name = "drugbank_id")
    private String drugbankId;

    @Column(name = "bigg_id")
    private Integer biggId;

    @Size(max = 255)
    @Column(name = "eafus_id")
    private String eafusId;

    @Size(max = 255)
    @Column(name = "knapsack_id")
    private String knapsackId;

    @Size(max = 255)
    @Column(name = "boiling_point")
    private String boilingPoint;

    @Size(max = 255)
    @Column(name = "boiling_point_reference")
    private String boilingPointReference;

    @Size(max = 255)
    @Column(name = "charge")
    private String charge;

    @Size(max = 255)
    @Column(name = "charge_reference")
    private String chargeReference;

    @Size(max = 255)
    @Column(name = "density")
    private String density;

    @Size(max = 255)
    @Column(name = "density_reference")
    private String densityReference;

    @Size(max = 255)
    @Column(name = "optical_rotation")
    private String opticalRotation;

    @Size(max = 255)
    @Column(name = "optical_rotation_reference")
    private String opticalRotationReference;

    @Size(max = 255)
    @Column(name = "percent_composition")
    private String percentComposition;

    @Size(max = 255)
    @Column(name = "percent_composition_reference")
    private String percentCompositionReference;

    @Lob
    @Column(name = "physical_description")
    private String physicalDescription;

    @Lob
    @Column(name = "physical_description_reference")
    private String physicalDescriptionReference;

    @Size(max = 255)
    @Column(name = "refractive_index")
    private String refractiveIndex;

    @Size(max = 255)
    @Column(name = "refractive_index_reference")
    private String refractiveIndexReference;

    @Size(max = 255)
    @Column(name = "uv_index")
    private String uvIndex;

    @Size(max = 255)
    @Column(name = "uv_index_reference")
    private String uvIndexReference;

    @Size(max = 255)
    @Column(name = "experimental_pka")
    private String experimentalPka;

    @Size(max = 255)
    @Column(name = "experimental_pka_reference")
    private String experimentalPkaReference;

    @Size(max = 255)
    @Column(name = "experimental_solubility_reference")
    private String experimentalSolubilityReference;

    @Size(max = 255)
    @Column(name = "experimental_logp_reference")
    private String experimentalLogpReference;

    @Size(max = 255)
    @Column(name = "hydrophobicity_reference")
    private String hydrophobicityReference;

    @Size(max = 255)
    @Column(name = "isoelectric_point_reference")
    private String isoelectricPointReference;

    @Size(max = 255)
    @Column(name = "melting_point_reference")
    private String meltingPointReference;

    @Size(max = 255)
    @Column(name = "moldb_alogps_logp")
    private String moldbAlogpsLogp;

    @Size(max = 255)
    @Column(name = "moldb_logp")
    private String moldbLogp;

    @Size(max = 255)
    @Column(name = "moldb_alogps_logs")
    private String moldbAlogpsLogs;

    @Lob
    @Column(name = "moldb_smiles")
    private String moldbSmiles;

    @Size(max = 255)
    @Column(name = "moldb_pka")
    private String moldbPka;

    @Size(max = 255)
    @Column(name = "moldb_formula")
    private String moldbFormula;

    @Column(name = "moldb_average_mass", precision = 9, scale = 4)
    private BigDecimal moldbAverageMass;

    @Lob
    @Column(name = "moldb_inchi")
    private String moldbInchi;

    @Column(name = "moldb_mono_mass", precision = 14, scale = 9)
    private BigDecimal moldbMonoMass;

    @Size(max = 255)
    @Column(name = "moldb_inchikey")
    private String moldbInchikey;

    @Size(max = 255)
    @Column(name = "moldb_alogps_solubility")
    private String moldbAlogpsSolubility;

    @Column(name = "moldb_id")
    private Integer moldbId;

    @Lob
    @Column(name = "moldb_iupac")
    private String moldbIupac;

    @Size(max = 255)
    @Column(name = "structure_source")
    private String structureSource;

    @Size(max = 255)
    @Column(name = "duplicate_id")
    private String duplicateId;

    @Size(max = 255)
    @Column(name = "old_dfc_id")
    private String oldDfcId;

    @Lob
    @Column(name = "dfc_name")
    private String dfcName;

    @Size(max = 255)
    @Column(name = "compound_source")
    private String compoundSource;

    @Size(max = 255)
    @Column(name = "flavornet_id")
    private String flavornetId;

    @Size(max = 255)
    @Column(name = "goodscent_id")
    private String goodscentId;

    @Size(max = 255)
    @Column(name = "superscent_id")
    private String superscentId;

    @Column(name = "phenolexplorer_metabolite_id")
    private Integer phenolexplorerMetaboliteId;

    @Size(max = 255)
    @Column(name = "kingdom")
    private String kingdom;

    @Size(max = 255)
    @Column(name = "superklass")
    private String superklass;

    @Size(max = 255)
    @Column(name = "klass")
    private String klass;

    @Size(max = 255)
    @Column(name = "subklass")
    private String subklass;

    @Size(max = 255)
    @Column(name = "direct_parent")
    private String directParent;

    @Size(max = 255)
    @Column(name = "molecular_framework")
    private String molecularFramework;

    @Size(max = 255)
    @Column(name = "chembl_id")
    private String chemblId;

    @Size(max = 255)
    @Column(name = "chemspider_id")
    private String chemspiderId;

    @Size(max = 255)
    @Column(name = "meta_cyc_id")
    private String metaCycId;

    @Column(name = "foodcomex")
    private Boolean foodcomex;

    @Size(max = 255)
    @Column(name = "phytohub_id")
    private String phytohubId;

    @Size(max = 255)
    @Column(name = "vmh_id")
    private String vmhId;

    @Size(max = 255)
    @Column(name = "fbonto_id")
    private String fbontoId;

    @NotNull
    @Column(name = "status", nullable = false)
    private Byte status;

}

File: DotNetServer/DotNetServer/Entities/Foodb/Compound.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Entities.Foodb;

[Table("compounds")]
[Index("Name", Name = "index_compounds_on_name", IsUnique = true)]
[Index("Name", "PublicId", Name = "index_compounds_on_name_and_public_id", IsUnique = true)]
[Index("PublicId", Name = "index_compounds_on_public_id", IsUnique = true)]
public partial class Compound
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("legacy_id")] public int? LegacyId { get; set; }

    [Column("type")] [StringLength(255)] public string Type { get; set; } = null!;

    [Column("public_id")]
    [StringLength(9)]
    public string PublicId { get; set; } = null!;

    [Column("name")] public string Name { get; set; } = null!;

    [Column("export")] public bool? Export { get; set; }

    [Column("state")] [StringLength(255)] public string? State { get; set; }

    [Column("annotation_quality")]
    [StringLength(255)]
    public string? AnnotationQuality { get; set; }

    [Column("description", TypeName = "mediumtext")]
    public string? Description { get; set; }

    [Column("cas_number")]
    [StringLength(255)]
    public string? CasNumber { get; set; }

    [Column("melting_point", TypeName = "mediumtext")]
    public string? MeltingPoint { get; set; }

    [Column("protein_formula")]
    [StringLength(255)]
    public string? ProteinFormula { get; set; }

    [Column("protein_weight")]
    [StringLength(255)]
    public string? ProteinWeight { get; set; }

    [Column("experimental_solubility")]
    [StringLength(255)]
    public string? ExperimentalSolubility { get; set; }

    [Column("experimental_logp")]
    [StringLength(255)]
    public string? ExperimentalLogp { get; set; }

    [Column("hydrophobicity")]
    [StringLength(255)]
    public string? Hydrophobicity { get; set; }

    [Column("isoelectric_point")]
    [StringLength(255)]
    public string? IsoelectricPoint { get; set; }

    [Column("metabolism", TypeName = "mediumtext")]
    public string? Metabolism { get; set; }

    [Column("kegg_compound_id")]
    [StringLength(255)]
    public string? KeggCompoundId { get; set; }

    [Column("pubchem_compound_id")]
    [StringLength(255)]
    public string? PubchemCompoundId { get; set; }

    [Column("pubchem_substance_id")]
    [StringLength(255)]
    public string? PubchemSubstanceId { get; set; }

    [Column("chebi_id")]
    [StringLength(255)]
    public string? ChebiId { get; set; }

    [Column("het_id")] [StringLength(255)] public string? HetId { get; set; }

    [Column("uniprot_id")]
    [StringLength(255)]
    public string? UniprotId { get; set; }

    [Column("uniprot_name")]
    [StringLength(255)]
    public string? UniprotName { get; set; }

    [Column("genbank_id")]
    [StringLength(255)]
    public string? GenbankId { get; set; }

    [Column("wikipedia_id")]
    [StringLength(255)]
    public string? WikipediaId { get; set; }

    [Column("synthesis_citations", TypeName = "mediumtext")]
    public string? SynthesisCitations { get; set; }

    [Column("general_citations", TypeName = "mediumtext")]
    public string? GeneralCitations { get; set; }

    [Column("comments", TypeName = "mediumtext")]
    public string? Comments { get; set; }

    [Column("protein_structure_file_name")]
    [StringLength(255)]
    public string? ProteinStructureFileName { get; set; }

    [Column("protein_structure_content_type")]
    [StringLength(255)]
    public string? ProteinStructureContentType { get; set; }

    [Column("protein_structure_file_size")]
    public int? ProteinStructureFileSize { get; set; }

    [Column("protein_structure_updated_at", TypeName = "datetime")]
    public DateTime? ProteinStructureUpdatedAt { get; set; }

    [Column("msds_file_name")]
    [StringLength(255)]
    public string? MsdsFileName { get; set; }

    [Column("msds_content_type")]
    [StringLength(255)]
    public string? MsdsContentType { get; set; }

    [Column("msds_file_size")] public int? MsdsFileSize { get; set; }

    [Column("msds_updated_at", TypeName = "datetime")]
    public DateTime? MsdsUpdatedAt { get; set; }

    [Column("creator_id")] public int? CreatorId { get; set; }

    [Column("updater_id")] public int? UpdaterId { get; set; }

    [Column("created_at", TypeName = "datetime")]
    public DateTime? CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime? UpdatedAt { get; set; }

    [Column("phenolexplorer_id")] public int? PhenolexplorerId { get; set; }

    [Column("dfc_id")] [StringLength(255)] public string? DfcId { get; set; }

    [Column("hmdb_id")]
    [StringLength(255)]
    public string? HmdbId { get; set; }

    [Column("duke_id")]
    [StringLength(255)]
    public string? DukeId { get; set; }

    [Column("drugbank_id")]
    [StringLength(255)]
    public string? DrugbankId { get; set; }

    [Column("bigg_id")] public int? BiggId { get; set; }

    [Column("eafus_id")]
    [StringLength(255)]
    public string? EafusId { get; set; }

    [Column("knapsack_id")]
    [StringLength(255)]
    public string? KnapsackId { get; set; }

    [Column("boiling_point")]
    [StringLength(255)]
    public string? BoilingPoint { get; set; }

    [Column("boiling_point_reference")]
    [StringLength(255)]
    public string? BoilingPointReference { get; set; }

    [Column("charge")] [StringLength(255)] public string? Charge { get; set; }

    [Column("charge_reference")]
    [StringLength(255)]
    public string? ChargeReference { get; set; }

    [Column("density")]
    [StringLength(255)]
    public string? Density { get; set; }

    [Column("density_reference")]
    [StringLength(255)]
    public string? DensityReference { get; set; }

    [Column("optical_rotation")]
    [StringLength(255)]
    public string? OpticalRotation { get; set; }

    [Column("optical_rotation_reference")]
    [StringLength(255)]
    public string? OpticalRotationReference { get; set; }

    [Column("percent_composition")]
    [StringLength(255)]
    public string? PercentComposition { get; set; }

    [Column("percent_composition_reference")]
    [StringLength(255)]
    public string? PercentCompositionReference { get; set; }

    [Column("physical_description", TypeName = "mediumtext")]
    public string? PhysicalDescription { get; set; }

    [Column("physical_description_reference", TypeName = "mediumtext")]
    public string? PhysicalDescriptionReference { get; set; }

    [Column("refractive_index")]
    [StringLength(255)]
    public string? RefractiveIndex { get; set; }

    [Column("refractive_index_reference")]
    [StringLength(255)]
    public string? RefractiveIndexReference { get; set; }

    [Column("uv_index")]
    [StringLength(255)]
    public string? UvIndex { get; set; }

    [Column("uv_index_reference")]
    [StringLength(255)]
    public string? UvIndexReference { get; set; }

    [Column("experimental_pka")]
    [StringLength(255)]
    public string? ExperimentalPka { get; set; }

    [Column("experimental_pka_reference")]
    [StringLength(255)]
    public string? ExperimentalPkaReference { get; set; }

    [Column("experimental_solubility_reference")]
    [StringLength(255)]
    public string? ExperimentalSolubilityReference { get; set; }

    [Column("experimental_logp_reference")]
    [StringLength(255)]
    public string? ExperimentalLogpReference { get; set; }

    [Column("hydrophobicity_reference")]
    [StringLength(255)]
    public string? HydrophobicityReference { get; set; }

    [Column("isoelectric_point_reference")]
    [StringLength(255)]
    public string? IsoelectricPointReference { get; set; }

    [Column("melting_point_reference")]
    [StringLength(255)]
    public string? MeltingPointReference { get; set; }

    [Column("moldb_alogps_logp")]
    [StringLength(255)]
    public string? MoldbAlogpsLogp { get; set; }

    [Column("moldb_logp")]
    [StringLength(255)]
    public string? MoldbLogp { get; set; }

    [Column("moldb_alogps_logs")]
    [StringLength(255)]
    public string? MoldbAlogpsLogs { get; set; }

    [Column("moldb_smiles", TypeName = "mediumtext")]
    public string? MoldbSmiles { get; set; }

    [Column("moldb_pka")]
    [StringLength(255)]
    public string? MoldbPka { get; set; }

    [Column("moldb_formula")]
    [StringLength(255)]
    public string? MoldbFormula { get; set; }

    [Column("moldb_average_mass")]
    [Precision(9, 4)]
    public decimal? MoldbAverageMass { get; set; }

    [Column("moldb_inchi", TypeName = "mediumtext")]
    public string? MoldbInchi { get; set; }

    [Column("moldb_mono_mass")]
    [Precision(14, 9)]
    public decimal? MoldbMonoMass { get; set; }

    [Column("moldb_inchikey")]
    [StringLength(255)]
    public string? MoldbInchikey { get; set; }

    [Column("moldb_alogps_solubility")]
    [StringLength(255)]
    public string? MoldbAlogpsSolubility { get; set; }

    [Column("moldb_id")] public int? MoldbId { get; set; }

    [Column("moldb_iupac", TypeName = "mediumtext")]
    public string? MoldbIupac { get; set; }

    [Column("structure_source")]
    [StringLength(255)]
    public string? StructureSource { get; set; }

    [Column("duplicate_id")]
    [StringLength(255)]
    public string? DuplicateId { get; set; }

    [Column("old_dfc_id")]
    [StringLength(255)]
    public string? OldDfcId { get; set; }

    [Column("dfc_name", TypeName = "mediumtext")]
    public string? DfcName { get; set; }

    [Column("compound_source")]
    [StringLength(255)]
    public string? CompoundSource { get; set; }

    [Column("flavornet_id")]
    [StringLength(255)]
    public string? FlavornetId { get; set; }

    [Column("goodscent_id")]
    [StringLength(255)]
    public string? GoodscentId { get; set; }

    [Column("superscent_id")]
    [StringLength(255)]
    public string? SuperscentId { get; set; }

    [Column("phenolexplorer_metabolite_id")]
    public int? PhenolexplorerMetaboliteId { get; set; }

    [Column("kingdom")]
    [StringLength(255)]
    public string? Kingdom { get; set; }

    [Column("superklass")]
    [StringLength(255)]
    public string? Superklass { get; set; }

    [Column("klass")] [StringLength(255)] public string? Klass { get; set; }

    [Column("subklass")]
    [StringLength(255)]
    public string? Subklass { get; set; }

    [Column("direct_parent")]
    [StringLength(255)]
    public string? DirectParent { get; set; }

    [Column("molecular_framework")]
    [StringLength(255)]
    public string? MolecularFramework { get; set; }

    [Column("chembl_id")]
    [StringLength(255)]
    public string? ChemblId { get; set; }

    [Column("chemspider_id")]
    [StringLength(255)]
    public string? ChemspiderId { get; set; }

    [Column("meta_cyc_id")]
    [StringLength(255)]
    public string? MetaCycId { get; set; }

    [Column("foodcomex")] public bool? Foodcomex { get; set; }

    [Column("phytohub_id")]
    [StringLength(255)]
    public string? PhytohubId { get; set; }

    [Column("vmh_id")] [StringLength(255)] public string? VmhId { get; set; }

    [Column("fbonto_id")]
    [StringLength(255)]
    public string? FbontoId { get; set; }

    [Column("status")] public sbyte Status { get; set; }

    [InverseProperty("Compound")]
    public virtual ICollection<CompoundAlternateParent> CompoundAlternateParents { get; set; } =
        new List<CompoundAlternateParent>();

    [InverseProperty("Compound")]
    public virtual ICollection<CompoundExternalDescriptor> CompoundExternalDescriptors { get; set; } =
        new List<CompoundExternalDescriptor>();

    [InverseProperty("Compound")]
    public virtual ICollection<CompoundSubstituent> CompoundSubstituents { get; set; } =
        new List<CompoundSubstituent>();

    [InverseProperty("Compound")]
    public virtual ICollection<CompoundsPathway> CompoundsPathways { get; set; } = new List<CompoundsPathway>();
}